### PR TITLE
fozzie-components@v3.44.0 - Bundlewatch optimisation #globalconfig

### PR DESCRIPTION
**Pull request by siggerzz**
This PR:

- Adds a new `maxBundleSize` property to each `package.json` file where we want Bundlewatch checks.
- Adds a new prompt to the component generator asking if Bundlewatch is required, and if so, what the max bundle size should be.

- Only runs Bundlewatch against changed packages if `maxBundleSize` is present.

### CODE REVIEW

**Comment by ashleynolan** [comment]
May need to update this a little as @xander-marjoram just merged in a PR on the bundlewatch config.

File: bundlewatch.config.js [hunk 1]
```diff
@@ -1,48 +1,44 @@
...
+const { execSync } = require('child_process');
+let outputChangedPackages;
+
+const maxBundleSizeExistsForPackage = packageLocation => {
+    const { maxBundleSize } = require(`${packageLocation}/package.json`);
+    
+    return (maxBundleSize);
+}
+
+const getMaxSizeForPackage = packageLocation => {
```
Comment (xander-marjoram) [inline bundlewatch.config.js:6]: What's the difference between these two methods?

File: bundlewatch.config.js [hunk 2]
```diff
@@ -1,48 +1,44 @@
...
+        outputChangedPackages = execSync('npx lerna ls --since origin/master --json');
+    } catch (error) {
+        console.info('No changed packages found.');
+        process.exit(0);
+    }
+
+    const changedPackagesArray = JSON.parse(outputChangedPackages.toString());
+
+    let changedPackageLocations = [];
+    changedPackagesArray.forEach(package => changedPackageLocations.push(package.location));
```
Comment (xander-marjoram) [inline bundlewatch.config.js]: Slight simplification
```suggestion
    const changedPackageLocations = changedPackagesArray.map(package => package.location);
```

File: packages/components/molecules/f-alert/package.json [hunk 3]
```diff
@@ -25,6 +26,7 @@
     "fozzie"
   ],
   "scripts": {
+
```
Comment (xander-marjoram) [inline packages/components/molecules/f-alert/package.json]: Just wondering if there's a reason for all these new lines?

File: packages/tools/generator-component/generators/app/prompts.js [hunk 4]
```diff
@@ -55,6 +55,20 @@ const prompts = [
...
         }
     },
+    {
+        message: 'Does your component require Bundlewatch checks?',
+        name: 'needsBundlewatch',
+        type: 'confirm',
+        default: true,
+    },
+    {
+        message: "What size limit do you want in place for your bundles (in kB)? e.g. '15'",
```
Comment (xander-marjoram) [inline packages/tools/generator-component/generators/app/prompts.js:65]: Is it possible to have a default value for this based on the type of component? Or set a very strict 5kB for new packages which can be increased later?

Comment (siggerzz) [reply packages/tools/generator-component/generators/app/prompts.js:65]: Will set 5kB as the default for now :)

**Review by xander-marjoram — APPROVED** [review]

**Comment by siggerzz** [comment]
Shouldn't have to pull master in this time around as there are no visual regression differences

**Review by DamianMullins — APPROVED** [review]

**Pull request merged**